enhancements/project-request-limit.md

```diff
@@ -0,0 +1,392 @@
+---
+title: Enabling ProjectRequestLimit on 4.x
+ 
+authors:
+  - "@akashem"
+
+reviewers:
+  - "@sttts"
+  - "@deads"
+  
+approvers:
+  - "@sttts"
+  - "@deads"    
+
+creation-date: 2020-03-03
+
+last-updated: 2020-03-03
+
+status: provisional
+
+see-also:
+
+replaces:
+
+superseded-by:
+
+---
+
+# Enabling ProjectRequestLimit on 4.x
+The `requestlimit.project.openshift.io/ProjectRequestLimit` admission plugin is not enabled in 4.x. The plugin already exists, this design 
+is about how we make it possible for a customer to enable it on a 4.x cluster.
+
+## Release Signoff Checklist
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+The `requestlimit.project.openshift.io/ProjectRequestLimit` admission plugin is used to impose a limit on the number of
+self-provisioned project(s) requested by a given user. In 3.x, the limit can be specified via the master configuration file.
+It doesn't fit well with our targeted configuration which aims to avoid adding lots of intricately documented knobs. 
+This is why this admission plug-in is disabled in OpenShift 4.x. 
+
+Instead of wiring the admission plugin via the `openshift-apiserver` operator, we can create a validating admission webhook 
+based on the [generic-admission-server](https://github.com/openshift/generic-admission-server) and distribute it via
+OperatorHub/OLM.
+
+## Motivation
+* Some customers use the `ProjectRequestLimit` plugin in 3.x and thus are blocked on their path to OpenShift 4.x. Enabling
+ it on 4.x will unblock these customers.
+* The `openshift-apiserver` (like any other apiserver in the system) is designed to be extended using validating/mutating 
+  admission webhook(s), we have the technology to easily build one, and we have the ability to create a simple operator to manage it.
+* The cluster admin can optionally enable the validating webhook on a cluster and configure limits via a custom
+resource. This will sever the "legacy" link between the plugin and the `kube-apiserver`.
```

Comment:
how is kube-apiserver involved?

Reply:
I keep forgetting, this used to be exposed via `openshift-apiserver`, fixed it.